x/dex/keeper/core.go

```diff
@@ -67,6 +67,10 @@ func (k Keeper) DepositCore(
 
 		lowerTick := k.GetOrInitTick(goCtx, pairId, lowerTickIndex)
 		upperTick := k.GetOrInitTick(goCtx, pairId, upperTickIndex)
+
+		sharesId := k.CreateSharesId(token0, token1, tickIndex, fee)
+		totalShares := k.bankKeeper.GetSupply(ctx, sharesId).Amount
+
 		pool, err := NewPool(
```

Comment:
Let's pass sharesId, totalShares into NewPool so they're stored on the pool

Reply:
see above